Utils/ViewExtensions.swift

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+extension UIView {
+    func screenshot(size: CGSize, offset: CGPoint? = nil, quality: Float = 1) -> UIImage? {
+        assert(0...1 ~= quality)
+
+        let offset = offset ?? CGPointMake(0, 0)
+
+        UIGraphicsBeginImageContextWithOptions(size, false, UIScreen.mainScreen().scale * CGFloat(quality))
+        drawViewHierarchyInRect(CGRect(origin: offset, size: frame.size), afterScreenUpdates: false)
+        var image = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+
+        return image
+    }
+
+    func screenshot(aspectRatio: Float? = nil, offset: CGPoint? = nil, quality: Float = 1) -> UIImage? {
```

Comment:
Ah here is the implementation :-) I would turn `aspectRatio` and `quality` into CGFloat types. Then you can get rid of all the casting that you do in both versions of `screenshot()` and things will be equal to UIKit graphics code.